File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {boolean}
     */
    containsNearbyDuplicate(nums, k) {
        if (nums.length <= 1) return false;
        let seen = new Set();
        let left = 0;
        for (let right = 0; right < nums.length; right++) {
            if (seen.has(nums[right])) return true;
            seen.add(nums[right]);
            if (right - left + 1> k) {
                seen.delete(nums[left]);
                left++;
            }

        }
        return false;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        left, right = 0, 0
        max_sum = nums[0]
        curr_sum = 0
        while right < len(nums):
            curr_sum = max(0, curr_sum)
            curr_sum += nums[right]
            max_sum = max(curr_sum, max_sum)

            if curr_sum < 0:
                left = right + 1
            right += 1
        return max_sum

            
